site/docs/releases.md

```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
   ...
 </dependencies>
 ```
+## 0.12.0 Release Notes
 
-## 0.11.1 release notes
+Apache Iceberg 0.12.0 was released on {{ DATE }}. It consists of {{ NUMCOMMITS }} commits contributed by {{ NUMCOMMITTERS }} over a {{ NUMDAYS }} day period.
+
+High-level features:
+
+* **Core**
+  * Allow Iceberg schemas to specify one or more columns as row identifers [[\#2465](https://github.com/apache/iceberg/pull/2465)]. Note that this is a prerequisite for supporting upserts in Flink.
+  * Added Jdbc [[\#1870](https://github.com/apache/iceberg/pull/1870)] and DynamoDB [[\#2688](https://github.com/apache/iceberg/pull/2688)] Iceberg catalog implementations.
```

Comment:
JDBC is normally all caps. Also, no need to qualify "catalog" as "Iceberg catalog".

Reply:
Done